app/controllers/katello/api/v2/content_view_components_controller.rb

```diff
@@ -28,6 +29,32 @@ def index_response
       }
     end
 
+    # content_views/:id/components/show_all
+    # Shows all content views, added and available to add, for a content view
+    # Undocumented endpoint since the functionality exists in separate calls already.
+    # This was created for ease of pagination and search for the UI
+    # param :id, :number, desc: N_("Content View id"), required: true
+    def show_all
+      kc = Katello::ContentView.table_name
+      kcc = Katello::ContentViewComponent.table_name
+      join_query = <<-SQL
+         LEFT OUTER JOIN #{kcc}
+         ON #{kc}.id = #{kcc}.content_view_id
+         AND #{kcc}.composite_content_view_id = #{@view.id}
+      SQL
+      order_query = <<-SQL
+         CAST (#{kcc}.composite_content_view_id as BOOLEAN) ASC, #{kc}.name
+      SQL
+
+      query = Katello::ContentView.readable.in_organization(@organization)
+      query = query&.non_composite&.non_default
+      custom_sort = ->(sort_query) { sort_query.joins(join_query).order(order_query) }
+      options = { resource_class: Katello::ContentView, custom_sort: custom_sort }
+      collection = scoped_search(query, nil, nil, options)
+      collection[:results] = ComponentViewPresenter.component_presenter(@view, views: collection[:results])
+      respond_for_index(:collection => collection, :template => "index")
```

Comment:
do you not need a custom rabl to include the existing 'content view component' rabl, but also an extra attribute to show if its in the content view for real or not?

Reply:
or maybe the existing rabl is enough? (for example id of the object would be nil).  Maybe being explicit is better, but curious your thoughts

Reply:
that works for me!  I'll test this api out a bit.  Code looks good